packages/flutter_tools/bin/xcode_backend.dart

```diff
@@ -350,16 +479,30 @@ class Context {
     }
   }
 
+  /// Calls `flutter assemble [buildMode]_[platform]_bundle_flutter_assets`
+  /// (e.g. `debug_ios_bundle_flutter_assets`, `debug_macos_bundle_flutter_assets`)
   void buildApp() {
     final bool verbose = (environment['VERBOSE_SCRIPT_LOGGING'] ?? '').isNotEmpty;
     final String sourceRoot = environment['SOURCE_ROOT'] ?? '';
     final String projectPath = environment['FLUTTER_APPLICATION_PATH'] ?? '$sourceRoot/..';
 
     final String buildMode = parseFlutterBuildMode();
+    final TargetPlatform platform = getPlatform();
 
-    final List<String> flutterArgs = _generateFlutterArgsForAssemble('build', buildMode, verbose);
+    final List<String> flutterArgs = _generateFlutterArgsForAssemble(
+      command: 'build',
+      buildMode: buildMode,
+      sourceRoot: sourceRoot,
+      platform: platform,
+      verbose: verbose,
+    );
 
-    flutterArgs.add('${buildMode}_ios_bundle_flutter_assets');
+    switch (platform) {
+      case TargetPlatform.ios:
+        flutterArgs.add('${buildMode}_ios_bundle_flutter_assets');
+      case TargetPlatform.macos:
+        flutterArgs.add('${buildMode}_macos_bundle_flutter_assets');
+    }
```

Comment:
Can this be simplified?

```suggestion
    flutterArgs.add('$buildMode_${platform.name}_bundle_flutter_assets');
```